projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -216,28 +264,87 @@ export class IgxDropDownComponent extends IgxDropDownBase implements IDropDownBa
 
     /**
      * Select an item by index
-     * @param index of the item to select
+     * @param index of the item to select; If the drop down uses *igxFor, pass the index in data
      */
     public setSelectedItem(index: number) {
         if (index < 0 || index >= this.items.length) {
             return;
         }
-        const newSelection = this.items[index];
+        let newSelection: IgxDropDownItemBase;
+        if (this.virtDir) {
+            newSelection = {
+                value: this.virtDir.igxForOf[index],
+                index
+            } as IgxDropDownItemBase;
+        } else {
+            newSelection = this.items[index];
+        }
         this.selectItem(newSelection);
     }
 
     /**
      * Navigates to the item on the specified index
+     * If the data in the drop-down is virtualized, pass the index of the item in the virtualized data.
      * @param newIndex number
      */
     public navigateItem(index: number) {
-        super.navigateItem(index);
+        if (this.virtDir) {
+            if (index === -1 || index >= this.collectionLength) {
+                return;
+            }
+            const virtState = this.virtDir.state;
+            const subRequired = index < virtState.startIndex || index >= virtState.chunkSize + virtState.startIndex - 1;
+            const direction = index > this._focusedIndex ? Navigate.Down : Navigate.Up;
+            if (subRequired) {
+                this.virtDir.scrollTo(index);
+            }
+            this._focusedIndex = index;
```

Comment:
Slight issue: `_focusedIndex` seems to be doing the same thing as `focusedItem` (sans the actual item), but the following skipHeader and other logic look for the item instead, so it won't work.